Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/Services/ComponentsController.cs

```diff
@@ -104,17 +106,27 @@ public HttpResponseMessage GetSuggestionRoles(string keyword, int roleGroupId, i
                     return this.Request.CreateResponse(HttpStatusCode.OK, new List<SuggestionDto>());
                 }
 
-                var matchedRoles = RoleController.Instance.GetRoles(this.PortalId)
-                    .Where(r => (roleGroupId == -2 || r.RoleGroupID == roleGroupId)
-                                && r.RoleName.IndexOf(keyword, StringComparison.InvariantCultureIgnoreCase) > -1
-                                   && r.Status == RoleStatus.Approved)
-                    .Select(r => new SuggestionDto()
+                var portalRoles = RoleController.Instance.GetRoles(this.PortalId);
+                
+
+                var matchedRoles = portalRoles.Where(r => (roleGroupId == -2 || r.RoleGroupID == roleGroupId)
+                                                          && r.RoleName.IndexOf(keyword, StringComparison.InvariantCultureIgnoreCase) > Null.NullInteger
+                                                          && r.SecurityMode != SecurityMode.SocialGroup
+                                                          && r.Status == RoleStatus.Approved).ToList();
+
+                if (roleGroupId <= Null.NullInteger
+                        && Globals.glbRoleUnauthUserName.IndexOf(keyword, StringComparison.InvariantCultureIgnoreCase) > Null.NullInteger)
+                {
+                    matchedRoles.Add(new RoleInfo { RoleID = int.Parse(Globals.glbRoleUnauthUser), RoleName = Globals.glbRoleUnauthUserName });
```

Comment:
Does `RoleName` need to be localized?

Reply:
We currently don't have a mechanism in DNN to locailze role names at any point. Admin's however, may rename default roles ("Administrators", "Registered Users")  if desired

Reply:
RoleID of special Roles should bestored in Portals record or  PortalSettings.